src/hoc/ClientSideStateUpdate.tsx

```diff
@@ -66,11 +66,12 @@ export const ClientSideStateUpdate = ({
 
     if (viewSettings) {
       store.dispatch(setViewSettings(viewSettings))
-      viewSettings.filterOptions?.type == FilterOptionsKeywords.CLIENTS
+      const view = viewSettingsTemp ? viewSettingsTemp.filterOptions : viewSettings.filterOptions
+      view?.type == FilterOptionsKeywords.CLIENTS
         ? store.dispatch(setFilteredAssgineeList({ filteredType: FilterByOptions.CLIENT }))
-        : viewSettings.filterOptions?.type == FilterOptionsKeywords.TEAM
+        : view?.type == FilterOptionsKeywords.TEAM
           ? store.dispatch(setFilteredAssgineeList({ filteredType: FilterByOptions.IUS }))
-          : viewSettings.filterOptions?.type == ''
+          : view.type == ''
             ? store.dispatch(setFilteredAssgineeList({ filteredType: FilterByOptions.NOFILTER }))
             : store.dispatch(setFilteredAssgineeList({ filteredType: FilterByOptions.NOFILTER }))
```

Comment:
```
? store.dispatch(setFilteredAssgineeList({ filteredType: FilterByOptions.NOFILTER }))
            : store.dispatch(setFilteredAssgineeList({ filteredType: FilterByOptions.NOFILTER }))
```

Aren't the expressions on the first and second part of the ternary operator doing the exact same thing? Why do we even need a check for `view.type == ''` in the first place then?